arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,1744 @@
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ReplicationProperties": {
+      "description": "The properties of a replication.",
+      "properties": {
+        "provisioningState": {
+          "description": "The status of the replication at the time the operation was called.",
+          "enum": [
+            "Creating",
+            "Succeeded"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "endpoint": {
+          "description": "The endpoint URI of the replication.",
+          "type": "string",
+          "readOnly": true
+        },
+        "isEnabled": {
+          "description": "The value that indicates whether the replication is enabled.",
+          "default": true,
+          "type": "boolean"
+        }
+      }
+    },
+    "ReplicationListResult": {
+      "description": "The result of a request to list replications for a container registry.",
+      "properties": {
+        "value": {
+          "description": "The list of replications. Since this list may be incomplete, the nextLink field should be used to request the next list of replications.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Replication"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next list of replications.",
+          "type": "string"
+        }
+      }
+    },
+    "Webhook": {
+      "description": "An object that represents a webhook for a container registry.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/WebhookProperties",
+          "description": "The properties of the webhook.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "WebhookProperties": {
+      "description": "The properties of a webhook.",
+      "properties": {
+        "isEnabled": {
+          "description": "The value that indicates whether the webhook is enabled.",
+          "default": true,
+          "type": "boolean"
+        },
+        "scope": {
+          "description": "The scope of repositories where the event can be triggered.",
+          "type": "string"
+        },
+        "events": {
+          "description": "The list of events that trigger the webhook to post notifications.",
+          "type": "array",
+          "items": {
+            "enum": [
+              "push",
+              "pull",
+              "delete"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "WebhookEvent",
+              "modelAsString": true
+            }
+          }
+        }
+      }
+    },
+    "WebhookCreateParameters": {
+      "description": "The parameters for creating a webhook.",
```

Comment:
This too, looks like a resource, simply do an allOf on ```Resource``` and remove the ```location``` property here